postnl_api/postnl_api.py

```diff
@@ -160,22 +160,20 @@ def _request_update(self, url, count=0, max=3):
         return response.json()
 
     def _request_login(self):
-        headers = {"Content-Type": "application/x-www-form-urlencoded"}
-
-        payload = {
-            "grant_type": "password",
-            "client_id": "pwAndroidApp",
```

Comment:
Would changing `pwAndroidApp` to `pwWebApp` not be enough? I am not sure why the whole call should be changed.

However if this fixes the problem, I am happy to merge and publish.

Reply:
I can confirm this fixes the autorisation issues in Home Assistant

Reply:
Is it possible to release this fix?

Reply:
@scornelissen85 sure! But to be honest I am not sure if this will fix it.

I did plan to do a full rewrite of the integration, however I have been more busy than expected lately, thus this will take a few weeks.

Were you able to confirm that this change still works?

Reply:
I have tried the fix, it will not work anymore.

Reply:
Mid April I had a report that this library still worked: https://github.com/ToonSoftwareCollective/postnl. Seeing their issue tracker however, it seems their login procedure also got broken recently :(

Reply:
> 
> 
> Mid April I had a report that this library still worked: https://github.com/ToonSoftwareCollective/postnl. Seeing their issue tracker however, it seems their login procedure also got broken recently :(
> 
> I'll check in with PostNL.

https://github.com/ToonSoftwareCollective/postnl I see that this has now been updated:  ### **fixed change logon procedure**

https://github.com/ToonSoftwareCollective/postnl/commit/14108d83a360ae9607d847858b4843b9d3bd8515

Can we use this fix also for our side?

Reply:
@iMicknl what are your thoughts?

Reply:
@peternijssen the approach mentioned above feels a bit hacky, but looks doable.

I just bought a house and I am in the planning and hopefully soon moving process, thus a bit more busy lately. However, this is certainly something I want for my new house ;-).

Reply:
GoGoGadget 💯  let's hope this can be implemented and can result again in a working PostNL function at home assistant 🤓

Reply:
I have tried the fix menionted, but that still does not work for me.
Also, the procedure is very different, it uses a shell script which places a cookie and scraped the html pages into tmp files.
The login procdure constist of a number of steps, with some generated tokens/codes/challenges and such.

A python script must probably do the same, which might be a challenge.

Reply:
@peternijssen did you get a reply back from PostNL? I had a look at the shell file and it doesn't work for me, and indeed as @Romkabouter mentions; they use scraping which I would like to avoid..